provisioning/transport/http/src/http.ts

```diff
@@ -0,0 +1,271 @@
+                callback(null, body);
+                this._fsm.transition('idle');
+              } else {
+                /* Codes_SRS_NODE_PROVISIONING_HTTP_18_030: [ If the operation status response has a success code with a 'status' that is not 'Assigned' or  'Assigning', `register` shall call the callback with a `SyntaxError` error and stop polling. ] */
+                this._safeCallback(callback, new SyntaxError('badly formed registration response'));
+                this._fsm.transition('idle');
+              }
+            });
+          },
+          register: (authorization, forceRegistration, body, callback) => {
+            /* Codes_SRS_NODE_PROVISIONING_HTTP_18_036: [ `register` shall call the `callback` with an `InvalidOperationError` if it is called while a previous registration is in progress. ] */
+            callback(new errors.InvalidOperationError('registration already in progress.'));
+          },
+          disconnect : (callback) => {
+            this._cancelCurrentOperation((err) => {
+              this._fsm.transition('idle');
+              this._safeCallback(callback);
+            });
+          }
+        }
+      }
+    });
+  }
+
+  register(authorization: string, forceRegistration: boolean, body: any, callback: ProvisioningResponseCallback): void {
+    /* Codes_SRS_NODE_PROVISIONING_HTTP_18_002: [ `register` shall throw a `ReferenceError` if `callback` is falsy. ] */
+    if (!callback) {
+      throw new ReferenceError();
+    }
+
+    this._fsm.handle('register', authorization, forceRegistration, body, callback);
+  }
+
+  disconnect(callback?: (err: Error) => void): void {
+    this._fsm.handle('disconnect', callback);
+  }
+
+  private _cancelCurrentOperation(callback: (err: Error) => void): void {
+    /* Codes_SRS_NODE_PROVISIONING_HTTP_18_034: [ disconnect will cancel any http operations that are in progress ] */
+    /* Codes_SRS_NODE_PROVISIONING_HTTP_18_035: [ disconnect will cause polling to cease ] */
+    if (this._currentHttpRequest != null) {
+      this._currentHttpRequest.abort();
+      this._currentHttpRequest = null;
+    }
+    if (this._pollingTimer != null) {
+      clearTimeout(this._pollingTimer);
+      this._pollingTimer = null;
+    }
+  }
+
+  private _sendRegistrationRequest(callback: (err: Error, body?: any) => void): void {
+    /* Codes_SRS_NODE_PROVISIONING_HTTP_18_009: [ `register` shall POST the registration request to 'https://global.azure-devices-provisioning.net/{idScope}/registrations/{registrationId}' ] */
+    /* Codes_SRS_NODE_PROVISIONING_HTTP_18_005: [ The registration request shall include the current `api-version` as a URL query string value named 'api-version'. ] */
+    let path: string = '/' + this._idScope + '/registrations/' + this._registrationId + '?api-version=' + this._apiVersion;
```

Comment:
>let path: string = '/' + this._idScope + '/registrations/' + this._registrationId + '?api-version=' + this._apiVersion; [](start = 4, length = 119)

looks like this path is the same in _sendRegistrationRequest and send OperationStatusRequest - maybe needs to be factored in somewhere.

Reply:
that was a typo.  the path was fixed.

---
In reply to: [141482380](https://github.com/Azure/azure-iot-sdk-node/pull/93#discussion_r141482380) [](ancestors = 141482380)